src/core/queryCache.js

```diff
@@ -396,7 +396,9 @@ export function makeQueryCache({ frozen = isServer, defaultConfig } = {}) {
               if (
                 isDocumentVisible() ||
                 query.instances.some(
-                  instance => instance.config.refetchIntervalInBackground
+                  instance =>
+                    instance.config.refetchIntervalInBackground &&
+                    instance.config.enabled
```

Comment:
This check needs to happen before the is document visible. If the query doesn’t have an enabled instance, then we should never even reach these other conditions.